TU20Bot/Handler.cs

```diff
@@ -155,50 +157,74 @@ public class Handler {
+                /*
+                 * This section handles the following cases:
+                 * Original message deleted:
+                 *          It should: Remove the event reference. Remove the event prompt.
+                 * Original message confirmed through prompt (The event prompt has been deleted):
+                 *          It should: Remove the reactions. Remove the event prompt.
+                 * Event prompt deleted before confirmation:
+                 *          This is an implied cancel. See below for details.
+                 * Event prompt canceled (❌) (The event prompt might has been deleted):
+                 *          It should: Remove the reactions. Remove the event reference. Remove the event prompt.
+                 */
+
                 var eventCollection = client.database.GetCollection<EventModel>(EventModel.collectionName);
-                
+                // Remove the bot reactions to the original message if possible.
                 // Let's delete any record of the event if the original message was deleted.
                 var model = await eventCollection.FindOneAndDeleteAsync(
-                    Builders<EventModel>.Filter.Eq(x => x.messageId, message.Id));
+                    Builders<EventModel>.Filter.Eq(x => x.messageId, deletedMessage.Id));
 
                 // Delete the prompt too, if it exists.
                 if (model?.promptId != null) {
                     await channel.DeleteMessageAsync(model.promptId.Value);
-                } else {
-                    // Otherwise, try to see if a prompt for a message was deleted.
+                } else if(model == null) {
+                    // When the flow reaches here then: the deletedMessage was not an indexed event (it is an event prompt or other)
+
+                    // Otherwise, try to see if an event prompt was deleted.
                     var promptModel = await eventCollection.FindOneAndUpdateAsync(
-                        Builders<EventModel>.Filter.Eq(x => x.promptId, message.Id),
+                        Builders<EventModel>.Filter.Eq(x => x.promptId, deletedMessage.Id),
                         Builders<EventModel>.Update.Set(x => x.promptId, null)
                     );
 
-                    // Then lets remove the bot reactions to the original message if possible.
-                    if (promptModel != null
-                        && await channel.GetMessageAsync(promptModel.messageId) is IUserMessage original) {
-                        // Some trickery to convert tag ids to emojis and vice-versa.
-                        var allTags = Tag.allTags.Select(x => new Emoji(x.emoji) as IEmote).ToArray();
+                    // If an event prompt was deleted,
+                    if(promptModel != null) {
+                        // If the event state is a draft (not confirmed), then the reference should be removed
+                        if(promptModel.state == EventState.Draft){
```

Comment:
```suggestion
                        if (promptModel.state == EventState.Draft){
```

Ahh this is becoming a pattern, there are other if statements that have this problem. Be careful with style, strange things like this can happen. It's worth looking at the other files in the project to decide on styling for your own contributions :)